plugin.video.catchuptvandmore/resources/lib/root/channels/fr/live_tv_fr.py

```diff
@@ -0,0 +1,399 @@
+    "Auvergne": "C1924.api.telerama.fr",
+    "Basse-Normandie": "C1925.api.telerama.fr",
+    "Bourgogne": "C1926.api.telerama.fr",
+    "Bretagne": "C1927.api.telerama.fr",
+    "Centre-Val de Loire": "C1928.api.telerama.fr",
+    "Chapagne-Ardenne": "C1929.api.telerama.fr",
+    "Corse": "C308.api.telerama.fr",
+    "Côte d'Azur": "C1931.api.telerama.fr",
+    "Franche-Compté": "C1932.api.telerama.fr",
+    "Haute-Normandie": "C1933.api.telerama.fr",
+    "Languedoc-Roussillon": "C1934.api.telerama.fr",
+    "Limousin": "C1935.api.telerama.fr",
+    "Lorraine": "C1936.api.telerama.fr",
+    "Midi-Pyrénées": "C1937.api.telerama.fr",
+    "Nord-Pas-de-Calais": "C1938.api.telerama.fr",
+    "Paris Île-de-France": "C1939.api.telerama.fr",
+    "Pays de la Loire": "C1940.api.telerama.fr",
+    "Picardie": "C1941.api.telerama.fr",
+    "Poitou-Charentes": "C1942.api.telerama.fr",
+    "Provence-Alpes": "C1943.api.telerama.fr",
+    "Rhône-Alpes": "C1944.api.telerama.fr"
+}
+
+LIVE_LA_1ERE = {
+    "Guadeloupe 1ère": "C329.api.telerama.fr",
+    "Guyane 1ère": "C260.api.telerama.fr",
+    "Martinique 1ère": "C328.api.telerama.fr",
+    "Mayotte 1ère": '',
+    "Nouvelle Calédonie 1ère": "C240.api.telerama.fr",
+    "Polynésie 1ère": "C459.api.telerama.fr",
+    "Réunion 1ère": "C245.api.telerama.fr",
+    "St-Pierre et Miquelon 1ère": '',
+    "Wallis et Futuna 1ère": ''
+}
+
+XMLTV_CHANNEL_ID = {
+    'tf1': 'C192.api.telerama.fr',
+    'france2': 'C4.api.telerama.fr',
+    'france3': 'C80.api.telerama.fr',
+    'france5': 'C47.api.telerama.fr',
+    'canalplus': 'C34.api.telerama.fr',
+    'c8': 'C445.api.telerama.fr',
+    'tmc': 'C195.api.telerama.fr',
+    'tfx': 'C446.api.telerama.fr',
+    'nrj12': 'C444.api.telerama.fr',
+    'france4': 'C78.api.telerama.fr',
+    'bfmtv': 'C481.api.telerama.fr',
+    'cnews': 'C226.api.telerama.fr',
+    'cstar': 'C458.api.telerama.fr',
+    'gulli': 'C482.api.telerama.fr',
+    'franceo': 'C160.api.telerama.fr',
+    'tf1-series-films': 'C1404.api.telerama.fr',
+    'lequipe': 'C1401.api.telerama.fr',
+    'numero23': 'C1402.api.telerama.fr',
+    'cherie25': 'C1399.api.telerama.fr',
+    'la_1ere': LIVE_LA_1ERE[common.PLUGIN.get_setting(
+        'la_1ere.region')],
+    'franceinfo': 'C2111.api.telerama.fr',
+    'bfmbusiness': 'C1073.api.telerama.fr',
+    'rmc': 'C546.api.telerama.fr',
+    'lci': 'C112.api.telerama.fr',
+    'lcp': 'C234.api.telerama.fr',
+    'rmcdecouverte': 'C1400.api.telerama.fr',
+    'publicsenat': '',
+    'francetvsport': 'multiple_streams',
+    'gong': '',
+    'france3regions': LIVE_FR3_REGIONS[common.PLUGIN.get_setting(
+        'france3.region')],
+    'bfmparis': ''
+}
+
+
+def download_xmltv():
+    if os.path.exists(XMLTV_FILEPATH):
+        mtime = os.stat(XMLTV_FILEPATH).st_mtime
+        dl_file = (time.time() - mtime > 3600)
+    else:
+        dl_file = True
+    if dl_file:
+        r = requests.get("https://repo.cecchettosylvain.fr/xmltv/xmltv_fr.xml")
```

Comment:
Where are you generating this XMLTV file from? XMLTV meta is licensed and usually only free to distribute under strict guidelines. ie if you are generating this via some type of unauthorized parsing script Kodi can not allow it. However if you are licensed to distribute this information or wrote a script to parse a websites meta to xmltv format you can include the parsing code in the script.

Reply:
I generate this XMLTV file on my personal server with a script that scrape data from a free French TV guide website.
Is it ok?

Reply:
I assumed this was the case...also noticed this while reviewing. We have a takedown policy anyway so even if this is violating some company terms of use they can always submit a complain.